### ISSUE TITLE
HSL/HSB/HSV Support

### ISSUE THREAD

**Issue by nasser** [nasser/zajal#39]
As mentioned [on the forums](http://forum.zajal.cc/discussion/12/hslhsvhsb-to-rgb), this is needed for badass coloring. Proposed syntax is

``` ruby
color_mode :hsb
color 23, 20, 80 # interpreted as HSB
```

Where the `color_mode` method takes symbols like `:hsb` and `:rgb` to change how the color method works.

The main open question is the ranges of the parameters for HS\* colors. Is 0-255 ok?

**Comment by nasser** [comment]
@judy is currently tackling this

**Comment by judy-zz** [comment]
I didn't even know we had forums. :-p

In the meanwhile, you can use the color interfaces already by pulling in the `color` branch. If you want to use HSV right now, you can do something like

``` ruby
red, green, blue = Color::HSV.new(0, 255, 255).to_rgb.to_a
color red, green, blue
```

All I have left to implement is color_mode, which would allow you to easily define the inputs that the `#color` method takes.

**Comment by nasser** [comment]
Ha! Yeah, I set them up at Parsons. Not too much activity. They'll probably get redone at some point.

I'll pull the color stuff in soon and build out the syntax for it.

Also, check this out! https://twitter.com/zajalcode/status/258767020756172800 http://logs.jruby.org/jruby/2012-10-18.html#T04-02-20  

R

**Comment by judy-zz** [comment]
That's really cool! Going pure Ruby was the right way to go; it seems to have opened quite a few doors. I can't wait to play around with more gems in Zajal too.

This is exciting stuff. We're on the right track. :-D  

Clinton Judy
judy.github.com (http://judy.github.com)

**Comment by judy-zz** [comment]
This is now done in my `color` branch, which I've sent a pull request for.

**Comment by nasser** [comment]
Fixed in 0b584c67
